lib/request.js

```diff
@@ -32,7 +34,14 @@ function buildRequest (R) {
 Object.defineProperties(Request.prototype, {
   req: {
     get () {
-      process.emitWarning('You are accessing the Node.js core request object via "request.req", Use "request.raw" instead.')
+      if (warningEmitted === true) {
+        return this.raw
+      }
+      warningEmitted = true
+      const warn = new Error('You are accessing the Node.js core request object via "request.req", Use "request.raw" instead.')
+      warn.name = 'FastifyDeprecation'
+      warn.code = 'FST_WARN'
```

Comment:
please use two separate codes

Reply:
I think we should follow core's example and enumerate all of our deprecations. So `warn.code` should include numeric suffix.

Reply:
Having separated codes also means that we should start tracking somewhere the codes we are using, which I fear might be overkill since we don't issue so many deprecations.

What do you think?

Reply:
I think it is better to establish the practice earlier rather than later. Maybe we never issue another deprecation ever again, but I suspect that is highly unlikely.

@fastify/core what are your opinions.